### PR TITLE
Adds an option to generate typecaches as zebras.

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
Adds an argument to typecache generation that allows specifying the whether to include/exclude types in the input list.
Also adds another argument to specify whether to remove falsey values after the typecache is generated.

## Why It's Good For The Game
Might make zaps slightly faster???
Honestly I just thought it would be a good way to condense some whitelist/blacklist typecache sets.

### CODE REVIEW

File: code/__HELPERS/_lists.dm [hunk 1]
```diff
@@ -155,16 +155,49 @@
...
+ * Ergo, subpaths should come before parent paths.
+ * Notice that this is the opposite priority of [/proc/typecacheof].
+ *
+ * Arguments:
+ * - path_to_check: A typepath to check.
+ * - [list_to_check][/list]: A list of typepaths to check the path_to_check against.
+ * - zebra: Whether to use the value of the mathing path in the list instead of just returning true when a match is found.
+ */
+/proc/is_path_in_list(path_to_check, list/list_to_check, zebra = FALSE)
+	if(!LAZYLEN(list_to_check) || !path_to_check)
```
Comment (MrMelbert) [inline code/__HELPERS/_lists.dm:193]:
```suggestion
	if(!LAZYLEN(list_to_check) || !ispath(path_to_check))
```
Might as well not bother iterating over the list if it's not a valid path in the first place

**Comment by Mothblocks** [comment]
This looks like it should be a separate proc as its fairly strongly different behavior as to what inputs its expecting

**Comment by TemporalOroboros** [comment]
> This looks like it should be a separate proc as its fairly strongly different behavior as to what inputs its expecting, rather than a boolean arg

Done.

**Comment by optimumtact** [comment]
you are getting absolutely owned by merge conflicts, ping me when you fix the next one

**Pull request merged**